src/main/java/uk/co/onsdigital/discovery/metadata/api/dao/MetadataDaoImpl.java

```diff
@@ -44,8 +63,26 @@ public DimensionalDataSet findDataSetById(String dataSetId) throws DataSetNotFou
     }
 
     @Override
+    public DimensionalDataSet findDataSetByEditionAndVersion(String dataResourceId, String edition, int version) throws DataSetNotFoundException {
+        final DimensionalDataSet dataSet =
+                entityManager.createNamedQuery(DimensionalDataSet.FIND_BY_EDITION_VERSION, DimensionalDataSet.class)
+                        .setParameter(DimensionalDataSet.DATA_RESOURCE_PARAM, dataResourceId)
+                        .setParameter(DimensionalDataSet.EDITION_PARAM, edition)
+                        .setParameter(DimensionalDataSet.VERSION_PARAM, version)
+                        .getSingleResult();
+        if (dataSet == null) {
```

Comment:
getSingleResult throws an exception if it can't find the result. You'll either need to trap that and convert it to a DatasetNotFoundException or query for a list and check whether the list is empty.
http://docs.oracle.com/javaee/7/api/javax/persistence/Query.html#getSingleResult--